### PR TITLE
Total assets

### PR DESCRIPTION
**Pull request by aziyatali**
Get total assets for each day!

### CODE REVIEW

File: src/kross-client/user.ts [hunk 1]
```diff
@@ -337,6 +337,30 @@ export class User extends KrossClientBase {
...
+          const accountLogs = await this.userAccountLogs({});
+          const noteLogs = await this.userNoteLogs({});
+          const accountLogsArray = Object.values(accountLogs?.data?.data);
+          const noteLogsArray = Object.values(noteLogs?.data?.data);
+          const totalAssets = {};
+          for (const accountLog of accountLogsArray) {
+            totalAssets[accountLog.save_date] = {
+              totalAsset: accountLog.amount,
+            };
+            for (const noteLog of noteLogsArray) {
```
Comment (Jscripter-pk) [inline src/kross-client/user.ts:351]: @aziyatali nested for loop should be avoided here.

File: src/kross-client/user.ts [hunk 2]
```diff
@@ -337,6 +337,30 @@ export class User extends KrossClientBase {
...
+              if (accountLog?.save_date === noteLog?.save_date) {
+                const totalAmount = accountLog.amount + noteLog.remain_principal;
+                totalAssets[accountLog.save_date] = {
+                  totalAsset: totalAmount,
+                };
+                break;
```
Comment (Jscripter-pk) [inline src/kross-client/user.ts:357]: Why we used break here?

Comment (aziyatali) [reply src/kross-client/user.ts:357]: skips the noteLog loop from and goes to next accountLog index

File: src/kross-client/user.ts (hunk 1 above)
Comment (aziyatali) [reply src/kross-client/user.ts:351]: Agreed, let me fix it.

File: src/kross-client/user.ts (hunk 2 above)
Comment (Jscripter-pk) [reply src/kross-client/user.ts:357]: instead of doing this way use JS built in functions like **find**

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]
@aziyatali Can you please cross check this with reference to frontend as from frontend we are showing based on selection filter so it will be better to search based on filter like 1month, 3 months, 6 months graph as shown in the design

**Comment by aziyatali** [comment]
> @aziyatali Can you please cross check this with reference to frontend as from frontend we are showing based on selection filter so it will be better to search based on filter like 1month, 3 months, 6 months graph as shown in the design

Makes sense rather getting whole data.

**Comment by aziyatali** [comment]
> @aziyatali Can you please cross check this with reference to frontend as from frontend we are showing based on selection filter so it will be better to search based on filter like 1month, 3 months, 6 months graph as shown in the design Screenshot 2023-02-06 at 11 59 22 AM also check for the percentage and attach it based on data from sdk

got it. Probably, (currentTotalAssets/xMonthsAgoTotalAssets) * 100

**Comment by aziyatali** [comment]
@Jscripter-pk  Now we can filter the date by inputing number, totalAssets(1) for 1 months etc. If data doesn't exist for that period then it return available data and growthRate

File: src/kross-client/user.ts [hunk 3]
```diff
@@ -337,6 +337,48 @@ export class User extends KrossClientBase {
...
+            totalAssets[accountLog.save_date] = {
+              totalAssets: accountLog.amount,
+            };
+          }
+          for (const noteLog of noteLogsArray) {
+            if (totalAssets[noteLog.save_date]) {
+              totalAssets[noteLog.save_date].totalAssets += noteLog.remain_principal;
+            }
+          }
+          const daysInMonth = 30 * number;
```
Comment (Jscripter-pk) [inline src/kross-client/user.ts:358]: @aziyatali this is not how we will calculate check the function below and return result based on date between start and end date 

```
 getStartAndEndDate = (
  repaymentFilter: string,
): [string, string] => {
  const month = repaymentFilter ? parseInt(repaymentFilter[0], 10) : 6;
  const today = new Date();
  const startDate = addMonths(today, -month);
  const endDate = addMonths(startDate, month);
  return [format(startDate, 'yyyy-MM-dd'), format(endDate, 'yyyy-MM-dd')];
};
```

Comment (Jscripter-pk) [reply src/kross-client/user.ts:358]: And also check by releasing version of it and test this by integrating in screen.

**Review by Jscripter-pk — CHANGES_REQUESTED** [review]

File: src/kross-client/user.ts (hunk 3 above)
Comment (aziyatali) [reply src/kross-client/user.ts:358]: Got it, could have considered that case

**Comment by aziyatali** [comment]
@Jscripter-pk Please review and merge if you think everything is okay. Tested with app and working properly. As you requested changed the argument range of dates.

**Review by Jscripter-pk — COMMENTED** [review]
@aziyatali push the integrations changes in this [PR](https://github.com/ninetydays/kross-investor-mobile-app/pull/76) with changes you tested

**Pull request merged**